front/lib/actions/mcp_internal_actions/servers/run_agent.ts

```diff
@@ -124,6 +128,33 @@ function createServer(auth: Authenticator): McpServer {
             return makeMCPToolTextError(errorMessage);
           } else if (event.type === "agent_message_success") {
             break;
+          } else if (event.type === "tool_approve_execution") {
+            if (_meta?.progressToken) {
+              await sendNotification({
+                method: "notifications/progress",
```

Comment:
Actually this is not really progress notifications, I should create something else? 🤔

Reply:
If I understand it correctly, notification method can be anything? 🤔 ihttps://github.com/modelcontextprotocol/modelcontextprotocol/blob/main/schema/2025-03-26/schema.ts#L50